src/components/SchemaRow.tsx

```diff
@@ -1,192 +1,67 @@
 import get = require('lodash/get');
-import map = require('lodash/map');
-import size = require('lodash/size');
 
-import { GoToRefHandler, IExtendableRenderers, SchemaNodeWithMeta, SchemaTreeListNode } from '../types';
-import { isCombiner, isRef } from '../utils';
-import { Types } from './';
+import { GoToRefHandler, SchemaNodeWithMeta, SchemaTreeListNode } from '../types';
+import { Caret } from './shared/Caret';
+import { Description } from './shared/Description';
+import { Property } from './shared/Property';
+import { Validations } from './shared/Validations';
 
-export interface ISchemaRow extends IExtendableRenderers {
+export interface ISchemaRow {
+  className?: string;
   node: SchemaTreeListNode;
   rowOptions: IRowRendererOptions;
   onGoToRef?: GoToRefHandler;
-  toggleExpand: () => void;
 }
 
 const ICON_SIZE = 12;
 const ICON_DIMENSION = 20;
+const ROW_OFFSET = 7;
 
-export const SchemaRow: React.FunctionComponent<ISchemaRow> = ({
-  node,
-  rowOptions,
-  onGoToRef,
-  rowRendererRight,
-  toggleExpand,
-}) => {
+export const SchemaRow: React.FunctionComponent<ISchemaRow> = ({ className, node, rowOptions, onGoToRef }) => {
   const schemaNode = node.metadata as SchemaNodeWithMeta;
-  const { name, $ref, subtype, required } = schemaNode;
-
-  const type = isRef(schemaNode) ? '$ref' : isCombiner(schemaNode) ? schemaNode.combiner : schemaNode.type;
   const description = get(schemaNode, 'annotations.description');
-  const childrenCount =
-    type === 'object'
-      ? size(get(schemaNode, 'properties'))
-      : subtype === 'object'
-        ? size(get(schemaNode, 'items.properties'))
-        : size(get(schemaNode, 'items'));
-
-  const nodeValidations = {
-    ...('annotations' in schemaNode && schemaNode.annotations.default
-      ? { default: schemaNode.annotations.default }
-      : {}),
-    ...get(schemaNode, 'validations', {}),
-  };
-  const validationCount = Object.keys(nodeValidations).length;
-  const handleGoToRef = React.useCallback<React.MouseEventHandler>(
-    () => {
-      if (onGoToRef) {
-        onGoToRef($ref!, node);
-      }
-    },
-    [onGoToRef, node, $ref],
-  );
-
-  const requiredElem = (
-    <div className={cn('ml-2', required ? 'font-medium' : 'text-darken-7 dark:text-lighten-6')}>
-      {required ? 'required' : 'optional'}
-      {validationCount ? `+${validationCount}` : ''}
-    </div>
-  );
 
-  const combinerOffset = ICON_DIMENSION * node.level;
   return (
-    <div onClick={toggleExpand} className="px-6 flex-1 w-full">
-      {/* Do not set position: relative. Divider must be relative to the parent container in order to avoid bugs related to this container calculated height changes. */}
+    <div className={cn('px-2 flex-1 w-full', className)}>
       <div
-        className="flex items-center text-sm"
+        className="flex items-center text-sm relative"
         style={{
-          marginLeft: combinerOffset,
+          marginLeft: ICON_DIMENSION * node.level, // offset for spacing
```

Comment:
Out of curiosity was there anything wrong with the previous implementation? Have you discovered bugs? Or is it a refactor?

Reply:
Returned back to the previous solution (pre-masking) that it's just more flexible.
The current version is kind of hacky as it relies on:
- padding (px-6)
- absolute positioning